transport/tchannel/handler.go

```diff
@@ -207,80 +221,84 @@ func (h handler) callHandler(ctx context.Context, call inboundCall, responseWrit
 	}
 }
 
-type responseWriter struct {
-	failedWith         error
-	format             tchannel.Format
-	headers            transport.Headers
-	buffer             *bufferpool.Buffer
-	response           inboundCallResponse
-	isApplicationError bool
-	headerCase         headerCase
+type tchannelResponseWriter struct {
+	FailedWith       error
+	Format           tchannel.Format
+	Headers          transport.Headers
+	Buffer           *bufferpool.Buffer
+	Response         inboundCallResponse
+	ApplicationError bool
+	HeaderCase       headerCase
```

Comment:
Do we need to change this struct at all?

Reply:
I believe so. From my understanding, the fields of tchannelResponseWriter need to be exposed as public in order to allow faultyResponseWriter struct{ tchannelResponseWriter } to embed its methods.